lib/dfe/analytics.rb

```diff
@@ -140,6 +158,16 @@ def self.entity_model_mapping
       # these back to table_names which are equivalent to dfe-analytics
       # "entities".
       @entity_model_mapping ||= begin
+        # Gems like devise put helper methods into controllers, and they add
+        # those methods at the time the routes file is parsed.
+        #
+        # Rails.configuration.eager_load = true will cause routes to be loaded
+        # before controllers; a direct call to Rails.application.eager_load! will
+        # not. To avoid this specific conflict with devise and possibly other
+        # gems/engines, proactively load the routes unless
+        # configuration.eager_load is set.
+        Rails.application.reload_routes! unless Rails.configuration.eager_load
```

Comment:
Did we only need to do this in `development` env? Or do we not need to check that because in non-dev the routes will be loaded at this point anyway? I don't quite remember what we said, but might be worth explaining why this is ok to do in production if there's anything  interesting to say.

Reply:
Will beef up this comment to explain what default configs there are in prod/dev